src/braket/circuits/gate_modifiers.py

```diff
@@ -0,0 +1,30 @@
+from functools import singledispatch
+from typing import List, Union
+
+import numpy as np
+
+ControlState = Union[int, List[int], str]
```

Comment:
That's a good point. `BasisState(state: BasisStateInput, size: Optional[int] = None)` with an `as_list` property could be a good fit.

I think it would also be fine to keep this as is, and if we want to later expand to other use cases, we could alias ControlState to point to a shared implementation. I like your suggestion though, so leaning towards making the switch now.